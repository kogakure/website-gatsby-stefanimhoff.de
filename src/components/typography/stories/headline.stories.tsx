import React from 'react';

import { Headline } from '..';

export default {
  title: 'Typography/Headline',
};

export const Default = () => <Headline>Headline</Headline>;

export const NoMargin = () => <Headline noMargin>Headline</Headline>;
