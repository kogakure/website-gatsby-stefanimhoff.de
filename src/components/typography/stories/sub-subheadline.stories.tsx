import React from 'react';

import { SubSubheadline } from '..';

export default {
  title: 'Typography/SubSubheadline',
};

export const Default = () => <SubSubheadline>SubSubheadline</SubSubheadline>;

export const NoMargin = () => (
  <SubSubheadline noMargin>SubSubheadline</SubSubheadline>
);
