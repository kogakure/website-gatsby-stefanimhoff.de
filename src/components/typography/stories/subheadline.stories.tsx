import React from 'react';

import { Subheadline } from '..';

export default {
  title: 'Typography/Subheadline',
};

export const Default = () => <Subheadline>Subheadline</Subheadline>;

export const NoMargin = () => <Subheadline noMargin>Subheadline</Subheadline>;
