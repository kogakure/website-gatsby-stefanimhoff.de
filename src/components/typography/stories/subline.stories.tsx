import React from 'react';

import { Subline } from '..';

export default {
  title: 'Typography/Subline',
};

export const Default = () => <Subline>Subline</Subline>;
