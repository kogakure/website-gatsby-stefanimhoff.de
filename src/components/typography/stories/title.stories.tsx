import React from 'react';

import { Title } from '..';

export default {
  title: 'Typography/Title',
};

export const Default = () => <Title>Title</Title>;

export const NoMargin = () => <Title noMargin>Title</Title>;
