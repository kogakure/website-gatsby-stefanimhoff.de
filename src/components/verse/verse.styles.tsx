import styled, { css } from 'styled-components';

import { space } from '../../layout/tokens';

import { VerseProps } from './verse';

export const variants = {
  center: {
    blockquote: css`
      margin: ${space[10]};
    `,
    pre: css`
      margin-left: auto;
      margin-right: auto;
    `,
  },
  left: {
    blockquote: css`
      margin-bottom: ${space[10]};
      margin-left: 0;
      margin-right: 0;
      margin-top: ${space[10]};
    `,
    pre: null,
  },
};

const Verse = styled.blockquote<VerseProps>`
  ${({ variant }) => variant && variants[variant].blockquote}
  display: flex;
  font-style: italic;

  p {
    margin: 0;
  }
`;

const Preformatted = styled.pre<VerseProps>`
  ${({ variant }) => variant && variants[variant].pre}
  font-family: inherit;
  margin-bottom: 0;
  margin-top: 0;
`;

export const Styled = {
  Preformatted,
  Verse,
};
