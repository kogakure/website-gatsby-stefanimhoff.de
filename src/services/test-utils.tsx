import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';

const customRender = (
  node: React.ReactElement,
  options?: RenderOptions
): RenderResult => {
  const rendered = render(
    <EmojifyProvider>
      <LocalizedDateProvider>{node}</LocalizedDateProvider>
    </EmojifyProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (ui: React.ReactElement, rerenderOptions?: RenderOptions) =>
      customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
