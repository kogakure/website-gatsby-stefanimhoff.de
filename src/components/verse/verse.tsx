import * as React from 'react';

import { Styled, variants } from './verse.styles';

export type VerseVariants = keyof typeof variants;
export type VerseProps = {
  variant?: VerseVariants;
};

export const Verse: React.FC<VerseProps> = ({
  variant = 'center',
  children,
  ...props
}) => (
  <Styled.Verse variant={variant} {...props}>
    <Styled.Preformatted variant={variant}>{children}</Styled.Preformatted>
  </Styled.Verse>
);
