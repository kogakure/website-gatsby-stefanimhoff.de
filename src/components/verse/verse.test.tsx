import React from 'react';
import { axe } from 'jest-axe';

import { render } from '../../services/test-utils';

import { Default, Center, Left } from './verse.stories';

describe('Verse', () => {
  test('renders correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('with "center" renders correctly', () => {
    const { container } = render(<Center />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('with "left" renders correctly', () => {
    const { container } = render(<Left />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('is accessible', async () => {
    const { container } = render(<Default />);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
