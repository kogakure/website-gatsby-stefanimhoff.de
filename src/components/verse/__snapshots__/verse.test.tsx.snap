// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verse renders correctly 1`] = `
.c0 {
  margin: clamp(1.44rem,3.70vw,1.728rem);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-style: italic;
}

.c0 p {
  margin: 0;
}

.c1 {
  margin-left: auto;
  margin-right: auto;
  font-family: inherit;
  margin-bottom: 0;
  margin-top: 0;
}

<blockquote
  class="c0"
>
  <pre
    class="c1"
  >
    <p>
      God guard me from those thoughts men think
      <br />
      In the mind alone;
      <br />
      He that sings a lasting song
      <br />
      Thinks in a marrow bone.
    </p>
  </pre>
</blockquote>
`;

exports[`Verse with "center" renders correctly 1`] = `
.c0 {
  margin: clamp(1.44rem,3.70vw,1.728rem);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-style: italic;
}

.c0 p {
  margin: 0;
}

.c1 {
  margin-left: auto;
  margin-right: auto;
  font-family: inherit;
  margin-bottom: 0;
  margin-top: 0;
}

<blockquote
  class="c0"
>
  <pre
    class="c1"
  >
    <p>
      God guard me from those thoughts men think
      <br />
      In the mind alone;
      <br />
      He that sings a lasting song
      <br />
      Thinks in a marrow bone.
    </p>
  </pre>
</blockquote>
`;

exports[`Verse with "left" renders correctly 1`] = `
.c0 {
  margin-bottom: clamp(1.44rem,3.70vw,1.728rem);
  margin-left: 0;
  margin-right: 0;
  margin-top: clamp(1.44rem,3.70vw,1.728rem);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-style: italic;
}

.c0 p {
  margin: 0;
}

.c1 {
  font-family: inherit;
  margin-bottom: 0;
  margin-top: 0;
}

<blockquote
  class="c0"
>
  <pre
    class="c1"
  >
    <p>
      God guard me from those thoughts men think
      <br />
      In the mind alone;
      <br />
      He that sings a lasting song
      <br />
      Thinks in a marrow bone.
    </p>
  </pre>
</blockquote>
`;
